framework/src/variables/MooseVariableData.C

```diff
@@ -654,11 +655,326 @@ MooseVariableData<OutputType>::computeValues()
     computeAD(num_dofs, nqp);
 }
 
+template <>
+void
+MooseVariableData<RealArrayValue>::computeValues()
```

Comment:
I thought of creating a derived Eigen class for ourselves and implement few more methods to avoid this duplication. But not sure if it is better or not.